src/test/java/org/jvnet/hudson/plugins/port_allocator/PortAllocationWorkflowTest.java

```diff
@@ -0,0 +1,105 @@
+package org.jvnet.hudson.plugins.port_allocator;
+
+import hudson.model.Label;
+import hudson.model.Node;
+import hudson.model.Result;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.slaves.DumbSlave;
+import hudson.slaves.NodeProperty;
+import hudson.slaves.RetentionStrategy;
+import org.jenkinsci.plugins.workflow.cps.CpsFlowDefinition;
+import org.jenkinsci.plugins.workflow.job.WorkflowJob;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.junit.Before;
+import org.junit.FixMethodOrder;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runners.MethodSorters;
+import org.jvnet.hudson.test.JenkinsRule;
+import org.jvnet.hudson.test.RestartableJenkinsRule;
+
+import java.util.Collections;
+import java.util.List;
+
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class PortAllocationWorkflowTest {
+
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Rule
+    public TemporaryFolder tmp = new TemporaryFolder();
+
+    @Test
+    public void wrap_01_WithNonExistingPool() throws Exception {
+        j.jenkins.addNode(new DumbSlave("slave", "dummy",
+            tmp.newFolder("remoteFS").getPath(), "1", Node.Mode.NORMAL, "",
+            j.createComputerLauncher(null), RetentionStrategy.NOOP, Collections.<NodeProperty<?>>emptyList()));
+        WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+            "node('slave') {\n"
+                + "  wrap([$class: 'PortAllocator', pools: ['WEBLOGIC']]) {\n"
```

Comment:
If you have a `@Symbol` you should not need to explicitly use the `wrap` step.

Reply:
@jglick can you give me an example plugin that does this?  trying to resurrect this PR

Reply:
never mind, think I found what I need: https://jenkins.io/doc/developer/plugin-development/pipeline-integration/

Reply:
@jglick I started resurrecting this pr: https://github.com/er1c/port-allocator-plugin/tree/workflow-support but I got stuck on this `@Symbol` thing.  

I get an error like:

```
java.lang.NoSuchMethodError: No such DSL method portallocator found among [archive, bat, build, catchError, checkout, dir, echo, error, fff, fileExists, injectSubtypesAsContext, input, load, mail, node, parallel, persistenceProblem, pwd, readFile, retry, sh, sleep, stage, step, timeout, tool, unarchive, waitUntil, withEnv, wrap, writeFile, ws]
	at org.jenkinsci.plugins.workflow.cps.DSL.invokeMethod(DSL.java:107)
	at org.jenkinsci.plugins.workflow.cps.CpsScript.invokeMethod(CpsScript.java:98)
	at org.codehaus.groovy.runtime.callsite.PogoMetaClassSite.call(PogoMetaClassSite.java:48)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:48)
```

Is this something you can take a quick look at?